x-pack/packages/security-solution/ecs_data_quality_dashboard/impl/data_quality/use_stats/index.test.tsx

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { renderHook } from '@testing-library/react-hooks';
+import React from 'react';
+
+import { DataQualityProvider } from '../data_quality_panel/data_quality_context';
+import { mockStatsGreenIndex } from '../mock/stats/mock_stats_green_index';
+import { ERROR_LOADING_STATS } from '../translations';
+import { useStats } from '.';
+
+const mockHttpFetch = jest.fn();
+const ContextWrapper: React.FC = ({ children }) => (
+  <DataQualityProvider httpFetch={mockHttpFetch}>{children}</DataQualityProvider>
+);
+
+const pattern = 'auditbeat-*';
+
+describe('useStats', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('happy path', () => {
+    beforeEach(() => {
+      mockHttpFetch.mockResolvedValue(mockStatsGreenIndex);
+    });
+
+    test('it returns the expected stats', async () => {
```

Comment:
Side question. I've seen it debated on a few PR's. Do you consider a benefit to having separate test blocks vs multiple expects in a single test block? Or even just testing the whole `result.current` object in one test?

Reply:
> Side question. I've seen it debated on a few PR's. Do you consider a benefit to having separate test blocks vs multiple expects in a single test block? Or even just testing the whole `result.current` object in one test?

I generally prefer to read and write "one assertion per test". That said, these tests had redundant  test setup code. Per our offline discussion, I refactored-out the redundant setup code, and applied this pattern to the other test files in this PR.